Cython/Compiler/Parsing.py

```diff
@@ -4313,11 +4313,11 @@ def p_mapping_pattern(s):
 
     double_star_capture_target = None
     items_patterns = []
-    double_star_set_twice = None
+    double_star_is_not_final = None
     while True:
+        if double_star_capture_target and not double_star_is_not_final:
+            double_star_is_not_final = s.position()
```

Comment:
`double_star_is_not_final` makes me think of the `final` C modifier. And we generally use `star_star` in other places. I suggest naming this variable `star_star_arg_pos`. (And the other variable accordingly.)